pkg/kv/kvserver/rditer/select_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package rditer
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/testutils/datapathutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/echotest"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSelect(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	w := echotest.NewWalker(t, datapathutils.TestDataPath(t, t.Name()))
+	for _, tc := range []struct {
+		name string
+		sp   roachpb.RSpan
+	}{
+		{
+			name: "no_span",
+		},
+		{
+			name: "r1",
+			sp: roachpb.RSpan{
+				// r1 is special - see https://github.com/cockroachdb/cockroach/issues/95055.
+				Key:    roachpb.RKeyMin,
+				EndKey: roachpb.RKey("c"),
+			},
+		},
+		{
+			name: "r2",
+			sp: roachpb.RSpan{
+				Key:    roachpb.RKey("a"),
+				EndKey: roachpb.RKey("c"),
+			},
+		},
```

Comment:
How about also a test with a span going to the end of the keyspace? Would it differ in any way from `r2`?

Reply:
It shouldn't! I added it to `TestSelect` and it performs as expected.